frontend/src/test/java/com/google/tripmeout/frontend/servlet/PlaceParentServletTest.java

```diff
@@ -0,0 +1,195 @@
+                                                  .setTripId("123abc")
+                                                  .setPlaceId("ChIJ3S-JXmauEmsRUcIaWtf4MzE")
+                                                  .setLatitude(111.111)
+                                                  .setLongitude(45.4)
+                                                  .setUserMark(PlaceVisitModel.UserMark.MAYBE)
+                                                  .build();
+
+  @Before
+  public void setUp() {
+    initMocks(this);
+  }
+
+  @Test
+  public void doPost_badUri_setsBadRequestStatus() throws IOException {
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+    PlaceVisitStorage placeStorage = new InMemoryPlaceVisitStorage();
+    PlaceParentServlet servlet = new PlaceParentServlet(placeStorage, gson);
+    when(request.getRequestURI()).thenReturn("/trips-number/abc123/placeVisits");
+    when(request.getReader())
+        .thenReturn(new BufferedReader(
+            new StringReader("{tripId: abc123, placeId: ChIJ3S-JXmauEmsRUcIaWtf4MzE}")));
+
+    servlet.doPost(request, response);
+
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_BAD_REQUEST);
+  }
+
+  @Test
+  public void doPost_badPlaceId_setsBadRequestStatus() throws IOException {
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+    PlaceVisitStorage placeStorage = new InMemoryPlaceVisitStorage();
+    PlaceParentServlet servlet = new PlaceParentServlet(placeStorage, gson);
+    when(request.getRequestURI()).thenReturn("/trips/abc123/placeVisits");
+    when(request.getReader())
+        .thenReturn(new BufferedReader(new StringReader("{tripId: abc123, placeId: hello}")));
+
+    servlet.doPost(request, response);
+    Optional<PlaceVisitModel> place = placeStorage.getPlaceVisit("abc123", "hello");
+
+    assertThat(place).isEmpty();
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_BAD_REQUEST);
+  }
+
+  @Test
+  public void doPost_goodRequest_storesPlaceVisitModel() throws IOException {
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+    PlaceVisitStorage placeStorage = new InMemoryPlaceVisitStorage();
+    PlaceParentServlet servlet = new PlaceParentServlet(placeStorage, gson);
+    when(request.getRequestURI()).thenReturn("/trips/abc123/placeVisits");
+    when(request.getReader())
+        .thenReturn(new BufferedReader(
+            new StringReader("{tripId: abc123, placeId: ChIJ3S-JXmauEmsRUcIaWtf4MzE}")));
+
+    servlet.doPost(request, response);
+    Optional<PlaceVisitModel> place =
+        placeStorage.getPlaceVisit("abc123", "ChIJ3S-JXmauEmsRUcIaWtf4MzE");
+
+    assertThat(place).isPresent();
+    assertThat(place.get().placeId()).isEqualTo("ChIJ3S-JXmauEmsRUcIaWtf4MzE");
+    assertThat(place.get().tripId()).isEqualTo("abc123");
+    assertThat(place.get().userMark()).isEqualTo(PlaceVisitModel.UserMark.YES);
```

Comment:
At a minimum, this should also assert that the response code is successful, otherwise we may miss certain server exceptions. Likewise whenever we expect a successful response and are not already checking the response code/response stream.